src/Spout/Writer/XLSX/Manager/Style/StyleRegistry.php

```diff
@@ -48,11 +121,58 @@ public function registerStyle(Style $style)
     {
         $registeredStyle = parent::registerStyle($style);
         $this->registerFill($registeredStyle);
+        $this->registerFormat($registeredStyle);
         $this->registerBorder($registeredStyle);
 
         return $registeredStyle;
     }
 
+    /**
+     * Register a format definition
+     *
+     * @param Style $style
+     */
+    protected function registerFormat(Style $style)
+    {
+        $styleId = $style->getId();
+
+        $format = $style->getFormat();
+        if ($format) {
+            $isFormatRegistered = isset($this->registeredFormats[$format]);
+
+            // We need to track the already registered format definitions
+            if ($isFormatRegistered) {
+                $registeredStyleId = $this->registeredFormats[$format];
+                $registeredFormatId = $this->styleIdToFormatsMappingTable[$registeredStyleId];
+                $this->styleIdToFormatsMappingTable[$styleId] = $registeredFormatId;
+            } else {
+                $this->registeredFormats[$format] = $styleId;
+
+                if (isset(self::$builtinNumFormatToIdMapping[$format])) {
+                    $id = self::$builtinNumFormatToIdMapping[$format];
+                } else {
+                    $id = $this->formatIndex++;
+                }
+                $this->styleIdToFormatsMappingTable[$styleId] = $id;
+            }
+        } else {
+            // The formatId maps a style to a format declaration
+            // When there is no format definition - we default to 0 ( General )
+            $this->styleIdToFormatsMappingTable[$styleId] = 0;
+        }
+    }
+
+    /**
+     * @param int $styleId
+     * @return int|null Format ID associated to the given style ID
+     */
+    public function getFormatIdForStyleId($styleId)
+    {
+        return (isset($this->styleIdToFormatsMappingTable[$styleId])) ?
```

Comment:
nitpick: can be simplified: `return $this->styleIdToFormatsMappingTable[$styleId] ?? null;`